pkg/asset/installconfig/aws/validation.go

```diff
@@ -49,12 +49,32 @@ func Validate(ctx context.Context, meta *Metadata, config *types.InstallConfig)
 	allErrs = append(allErrs, validatePlatform(ctx, meta, field.NewPath("platform", "aws"), config.Platform.AWS, config.Networking, config.Publish)...)
 
 	if config.ControlPlane != nil && config.ControlPlane.Platform.AWS != nil {
-		allErrs = append(allErrs, validateMachinePool(ctx, meta, field.NewPath("controlPlane", "platform", "aws"), config.Platform.AWS, config.ControlPlane.Platform.AWS, controlPlaneReq)...)
+		allErrs = append(allErrs, validateMachinePool(ctx, meta, field.NewPath("controlPlane", "platform", "aws"), config.Platform.AWS, config.ControlPlane.Platform.AWS, controlPlaneReq, "")...)
 	}
+
 	for idx, compute := range config.Compute {
 		fldPath := field.NewPath("compute").Index(idx)
+
 		if compute.Platform.AWS != nil {
-			allErrs = append(allErrs, validateMachinePool(ctx, meta, fldPath.Child("platform", "aws"), config.Platform.AWS, compute.Platform.AWS, computeReq)...)
+			switch compute.Name {
+			// Pool's specific validation
+			// Edge Compute Pool: AWS Local Zones is valid only when installing in existing VPC.
+			case types.MachinePoolEdgeRoleName:
```

Comment:
Is there an intention to add more cases? If not then this could be an if right ?

Reply:
No, if we haven't any checks specific for `worker` compute type. As it's not the case, I will replace the `switch` here. Thanks!